packages/graph/hash_graph/lib/graph/src/types/schema/data.rs

```diff
@@ -0,0 +1,316 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+use crate::types::schema::{Uri, Validate, ValidationError};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(
+    try_from = "DataTypeRepr",
+    tag = "kind",
+    rename = "dataType",
+    rename_all = "camelCase"
+)]
+pub struct DataType {
+    #[serde(rename = "$id")]
+    id: Uri,
+    title: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Do we want to consider [using this](https://docs.rs/serde_with/latest/serde_with/attr.skip_serializing_none.html) at some point perhaps

Reply:
I don't really see the benefit in this. Our serde definitions are pretty straight forward and using `serde_with` doesn't make it more readable. Thoughts @thehabbos007 ?

Reply:
I think for this one we won't. I guess my expectation was that data-types will have a lot more fields (we want to move away from `additional_properties`. We can revisit if we have more of these statements. Definitely don't think it's worth adding another dependency (especially with macros) at this point. Was just flagging its existence :)

Reply:
I don't see the benefit now either, but appreciate the flagging, it would clear up the struct definition down the line if we have lots of optional fields, but agree with alfie that it's not worth adding another dependency for it right now

Reply:
Resolving as I think enough of us are aware of it, so we can find it in the future if we need it 😁